run_incremental_zephir.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require "date"
+require "logger"
+
+logger = Logger.new($stdout)
+HOME = File.expand_path(__dir__).freeze
+INCREMENTAL_SCRIPT = File.join(HOME, "run_process_zephir_incremental.sh")
+
+# Get all of the Zephir files in /htsolr/catalog/prep/ (CATALOG_PREP)
+# of type zephir_upd_YYYYMMDD.json.gz
+all_zephir_upd = Dir.glob(File.join(ENV["CATALOG_PREP"], "zephir_upd_*")).map do |path|
```

Comment:
`zephir_upd_YYYYMMDD.json.gz` is (one of) the outputs, so we probably need to check which **don't** exist (up to the current date) and handle those... It gets somewhat complicated because we might have downloaded the file from Zephir but not finished processing all the outputs, or we might not have downloaded the file from Zephir at all...

The README has some fairly good documentation on what the files that are inputs and outputs.. but where they end up at various points in the process is less well-documented.

Reply:
The glob part of the new code is for finding what we have; for finding what we still need the logic around line 26 calculates the delta. It would be clearer (not to mention more correct) if the `else` condition were rewritten `dates = ((all_zephir_upd.last + 1)..(Date.today - 1)).to_a.sort`.

Even though this repo isn't ready for a full Ruby-ization, I'll see if I can extract this into `lib/` and add some functionality for looking at some of the other exports like the `.rights` file.

Reply:
Got it, I think I didn't understand what line 26 was doing.

One case to think about (maybe not for this iteration) is what happens if things run to completion on day N, fail on day N+1, and then work on day N+2. As written, I think this would start running at day N+3 (the day after the last complete update), but I think in reality we would need to go back and process day N+1 followed by day N+2 up through the present (because things need to happen in order...)

Reply:
Looking at a strategy of finding the "earliest gap" where there's one or more days missing from the output directories of interest. And not trying to fill just the gaps (because of deletes) but essentially running back the head and replaying. I'm going on the assumption that the scan for missing dates need not cross month boundaries (e.g. if it is the first of the month on the calendar, the target datestamp is last month so only need to scan thru the files with that month in the datestamp looking for gaps).

Reply:
That seems right to me.

Reply:
I'm hoping that `PostZephirProcessing::MonthlyInventory` is a better-designed solution. There are some things I would like to do differently once the constraint of using vanilla Ruby is lifted, but I'm reasonably happy with what is here.